course/page/base.py

```diff
@@ -765,9 +767,11 @@ def __init__(self, vctx, location, page_desc):
             try:
                 md_body = self.markup_body_for_title()
             except NotImplementedError:
-                from warnings import warn
-                warn(_("PageBaseWithTitle subclass '%s' does not implement "
-                        "markdown_body_for_title()")
+                if vctx is not None:
+                    vctx.add_warning(
+                        location,
+                        _("PageBaseWithTitle subclass '%s' does not implement "
+                          "markdown_body_for_title()")
```

Comment:
Same.

Reply:
Also, that warning gets the method name wrong. (Not introduced by you, but worth fixing nonetheless.)

Reply:
> that warning gets the method name wrong

Will fix that.